plugins/block-sharable-procedures/src/events_procedure_parameter_base.ts

```diff
@@ -30,12 +30,43 @@ export abstract class ProcedureParameterBase extends ProcedureBase {
    * Returns true if the given parameter is identical to this parameter.
    * @param param The parameter to check for equivalence.
    * @returns True if the parameter matches, false if it does not.
+   * @internal
    */
   protected parameterMatches(
       param?: Blockly.procedures.IParameterModel): boolean {
     return param && param.getId() === this.parameter.getId();
   }
 
+  /**
+   * Finds the parameter with the given ID in the procedure model with the given
+   * ID, if both things exist.
+   * @param workspace The workspace to search for the parameter.
+   * @param modelId The ID of the model to search for the parameter.
```

Comment:
Naming: what is this "model" whose ID you speak of?  Is this a procedure model?  Does the procedure model potentially have a different ID than the procedure it models?  If not, would `procedureId` be a better name?